### ISSUE TITLE
linter: False positive from no-useless-assignment

### ISSUE THREAD

**Issue by crystalfp** [oxc-project/oxc#21576]
### What version of Oxlint are you using?

1.60.0

### What command did you run?

`oxlint -c ./oxlint.json --tsconfig=tsconfig.json`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
    "$schema": "./node_modules/oxlint/configuration_schema.json",
  "categories": {
    "correctness": "warn",
    "suspicious": "warn",
    "perf": "warn",
    "pedantic": "warn",
    "nursery": "warn"
  },
  "plugins": [
    "vue",
    "import",
    "eslint",
    "typescript",
    "promise",
    "node",
    "unicorn",
    "oxc"
  ],
  "jsPlugins": [
    "eslint-plugin-sonarjs",
    "@stylistic/eslint-plugin",
    "@eslint-community/eslint-plugin-eslint-comments",
    "eslint-plugin-security",
    "eslint-plugin-regexp",
    "eslint-plugin-depend",
    "eslint-plugin-tsdoc"
  ],
  "ignorePatterns": [
    "src/vite-env.d.ts",
    "src/assets",
    "src/cpp",
    "src/electron/fingerprint/rollup.config.mjs",
    "src/electron/fingerprint/KDtree.js",
    "save/**",
    "proto-test/**",
    "docgen.config.js",
    "eslint.config.mjs",
    "eslint.vue.config.mjs",
    ".dependency-cruiser.js",
    "work/**",
    "public/doc/scripts.js",
    "tests/**"
  ],
  "options":{
    "typeAware": true,
    "typeCheck": true
  },
  "env": {
    "es2026": true,
    "builtin": true
  },
  "rules": {
    "constructor-super": "error",
    "for-direction": "error",
    "getter-return": "error",
    "no-async-promise-executor": "error",
    "no-case-declarations": "error",
    "no-class-assign": "error",
    "no-compare-neg-zero": "error",
    "no-cond-assign": "error",
    "no-const-assign": "error",
    "no-constant-binary-expression": "error",
    "no-constant-condition": "error",
    "no-control-regex": "error",
    "no-debugger": "error",
    "no-delete-var": "error",
    "no-dupe-class-members": "error",
    "no-dupe-else-if": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "no-empty": "error",
    "no-empty-character-class": "off",
    "no-empty-pattern": "error",
    "no-empty-static-block": "error",
    "no-ex-assign": "error",
    "no-extra-boolean-cast": "error",
    "no-fallthrough": "error",
    "no-func-assign": "error",
    "no-global-assign": "error",
    "no-import-assign": "error",
    "no-invalid-regexp": "off",
    "no-irregular-whitespace": "error",
    "no-loss-of-precision": "error",
    "no-misleading-character-class": "error",
    "no-new-native-nonconstructor": "error",
    "no-nonoctal-decimal-escape": "error",
    "no-obj-calls": "error",
    "no-prototype-builtins": "error",
    "no-redeclare": ["error", {"builtinGlobals": true}],
    "no-regex-spaces": "error",
    "no-self-assign": "error",
    "no-setter-return": "error",
    "no-shadow-restricted-names": "error",
    "no-sparse-arrays": "error",
    "no-this-before-super": "error",
    "no-undef": "error",
    "no-unexpected-multiline": "error",
    "no-unreachable": "error",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "error",
    "no-unsafe-optional-chaining": "error",
    "no-unused-labels": "error",
    "no-unused-private-class-members": "error",
    "no-unused-vars": [
      "error",
      {
        "vars": "all",
        "args": "all",
        "argsIgnorePattern": "^_",
        "caughtErrors": "all",
        "caughtErrorsIgnorePattern": "^_",
        "destructuredArrayIgnorePattern": "^_",
        "varsIgnorePattern": "^_",
        "ignoreRestSiblings": true
      }
    ],
    "no-useless-backreference": "off",
    "no-useless-catch": "error",
    "no-useless-escape": "error",
    "no-with": "error",
    "require-yield": "error",
    "use-isnan": "error",
    "valid-typeof": "error",
    "promise/always-return": "off",
    "promise/no-return-wrap": "error",
    "promise/param-names": "error",
    "promise/catch-or-return": [
      "warn",
      {
        "allowFinally": true
      }
    ],
    "promise/no-nesting": "warn",
    "promise/no-promise-in-callback": "warn",
    "promise/no-callback-in-promise": "warn",
    "promise/avoid-new": "off",
    "promise/no-new-statics": "error",
    "promise/no-return-in-finally": "warn",
    "promise/valid-params": "warn",
    "import/named": "off",
    "import/namespace": "error",
    "import/default": "off",
    "import/export": "error",
    "import/no-named-as-default": "warn",
    "import/no-named-as-default-member": "warn",
    "import/no-duplicates": "warn",
    "no-negated-condition": "warn",
    "no-nested-ternary": "off",
    "unicorn/catch-error-name": "error",
    "unicorn/consistent-assert": "error",
    "unicorn/consistent-date-clone": "error",
    "unicorn/consistent-empty-array-spread": "error",
    "unicorn/consistent-existence-index-check": "error",
    "unicorn/consistent-function-scoping": "error",
    "unicorn/empty-brace-spaces": "error",
    "unicorn/error-message": "error",
    "unicorn/escape-case": "error",
    "unicorn/explicit-length-check": "error",
    "unicorn/filename-case": [
      "off",
      {
        "case": "camelCase"
      }
    ],
    "unicorn/no-length-as-slice-end": "error",
    "unicorn/new-for-builtins": "off",
    "unicorn/no-abusive-eslint-disable": "error",
    "unicorn/no-accessor-recursion": "error",
    "unicorn/no-anonymous-default-export": "error",
    "unicorn/no-array-callback-reference": "error",
    "unicorn/no-array-for-each": "error",
    "unicorn/no-array-method-this-argument": "error",
    "unicorn/no-array-reduce": "error",
    "unicorn/no-array-reverse": "error",
    "unicorn/no-array-sort": "error",
    "unicorn/no-await-expression-member": "error",
    "unicorn/no-await-in-promise-methods": "error",
    "unicorn/no-console-spaces": "error",
    "unicorn/no-document-cookie": "error",
    "unicorn/no-empty-file": "error",
    "unicorn/no-hex-escape": "error",
    "unicorn/no-instanceof-builtins": "error",
    "unicorn/no-invalid-fetch-options": "error",
    "unicorn/no-invalid-remove-event-listener": "error",
    "unicorn/no-lonely-if": "error",
    "unicorn/no-magic-array-flat-depth": "error",
    "unicorn/no-negated-condition": "error",
    "unicorn/no-negation-in-equality-check": "error",
    "unicorn/no-nested-ternary": "error",
    "unicorn/no-new-array": "error",
    "unicorn/no-new-buffer": "error",
    "unicorn/no-null": "error",
    "unicorn/no-object-as-default-parameter": "error",
    "unicorn/no-process-exit": "error",
    "unicorn/no-single-promise-in-promise-methods": "error",
    "unicorn/no-static-only-class": "error",
    "unicorn/no-thenable": "error",
    "unicorn/no-this-assignment": "error",
    "unicorn/no-typeof-undefined": "error",
    "unicorn/no-unnecessary-array-flat-depth": "error",
    "unicorn/no-unnecessary-array-splice-count": "error",
    "unicorn/no-unnecessary-await": "error",
    "unicorn/no-unnecessary-slice-end": "error",
    "unicorn/no-unreadable-array-destructuring": "error",
    "unicorn/no-unreadable-iife": "error",
    "unicorn/no-useless-collection-argument": "error",
    "unicorn/no-useless-error-capture-stack-trace": "error",
    "unicorn/no-useless-fallback-in-spread": "error",
    "unicorn/no-useless-length-check": "error",
    "unicorn/no-useless-promise-resolve-reject": "error",
    "unicorn/no-useless-spread": "error",
    "unicorn/no-useless-switch-case": "error",
    "unicorn/no-useless-undefined": "error",
    "unicorn/no-zero-fractions": "off",
    "unicorn/number-literal-case": "error",
    "unicorn/numeric-separators-style": [
      "off",
      {
        "number": {
          "onlyIfContainsSeparator": true,
          "minimumDigits": 3
        }
      }
    ],
    "unicorn/prefer-add-event-listener": "error",
    "unicorn/prefer-array-find": "error",
    "unicorn/prefer-array-flat": "error",
    "unicorn/prefer-array-flat-map": "error",
    "unicorn/prefer-array-index-of": "error",
    "unicorn/prefer-array-some": "error",
    "unicorn/prefer-at": "error",
    "unicorn/prefer-bigint-literals": "error",
    "unicorn/prefer-blob-reading-methods": "error",
    "unicorn/prefer-class-fields": "warn",
    "unicorn/prefer-classlist-toggle": "error",
    "unicorn/prefer-code-point": "error",
    "unicorn/prefer-date-now": "error",
    "unicorn/prefer-dom-node-append": "error",
    "unicorn/prefer-dom-node-dataset": "error",
    "unicorn/prefer-dom-node-remove": "error",
    "unicorn/prefer-dom-node-text-content": "error",
    "unicorn/prefer-event-target": "error",
    "unicorn/prefer-global-this": "error",
    "unicorn/prefer-includes": "error",
    "unicorn/prefer-logical-operator-over-ternary": "error",
    "unicorn/prefer-math-min-max": "error",
    "unicorn/prefer-math-trunc": "error",
    "unicorn/prefer-modern-dom-apis": "error",
    "unicorn/prefer-modern-math-apis": "error",
    "unicorn/prefer-native-coercion-functions": "error",
    "unicorn/prefer-negative-index": "error",
    "unicorn/prefer-node-protocol": "error",
    "unicorn/prefer-number-properties": "error",
    "unicorn/prefer-object-from-entries": "error",
    "unicorn/prefer-optional-catch-binding": "error",
    "unicorn/prefer-prototype-methods": "error",
    "unicorn/prefer-query-selector": "error",
    "unicorn/prefer-reflect-apply": "error",
    "unicorn/prefer-regexp-test": "error",
    "unicorn/prefer-response-static-json": "error",
    "unicorn/prefer-set-has": "error",
    "unicorn/prefer-set-size": "error",
    "unicorn/prefer-spread": "error",
    "unicorn/prefer-string-raw": "error",
    "unicorn/prefer-string-replace-all": "error",
    "unicorn/prefer-string-slice": "error",
    "unicorn/prefer-string-starts-ends-with": "error",
    "unicorn/prefer-string-trim-start-end": "error",
    "unicorn/prefer-structured-clone": "error",
    "unicorn/prefer-top-level-await": "off",
    "unicorn/prefer-type-error": "error",
    "unicorn/require-array-join-separator": "error",
    "unicorn/require-module-specifiers": "error",
    "unicorn/require-number-to-fixed-digits-argument": "error",
    "unicorn/require-post-message-target-origin": "error",
    "unicorn/switch-case-braces": "off",
    "unicorn/text-encoding-identifier-case": "error",
    "unicorn/throw-new-error": "off",
    "@typescript-eslint/ban-ts-comment": [
      "error",
      {
        "minimumDescriptionLength": 10
      }
    ],
    "no-array-constructor": "error",
    "@typescript-eslint/no-duplicate-enum-values": "error",
    "@typescript-eslint/no-empty-object-type": "error",
    "@typescript-eslint/no-explicit-any": "error",
    "@typescript-eslint/no-extra-non-null-assertion": "error",
    "@typescript-eslint/no-misused-new": "error",
    "@typescript-eslint/no-namespace": "error",
    "@typescript-eslint/no-non-null-asserted-optional-chain": "error",
    "@typescript-eslint/no-require-imports": "error",
    "@typescript-eslint/no-this-alias": "error",
    "@typescript-eslint/no-unnecessary-type-constraint": "error",
    "@typescript-eslint/no-unsafe-declaration-merging": "error",
    "@typescript-eslint/no-unsafe-function-type": "error",
    "no-unused-expressions": "error",
    "@typescript-eslint/no-wrapper-object-types": "error",
    "@typescript-eslint/prefer-as-const": "error",
    "@typescript-eslint/prefer-optional-chain": "error",
    "@typescript-eslint/prefer-namespace-keyword": "error",
    "@typescript-eslint/triple-slash-reference": "error",
    "@typescript-eslint/await-thenable": "error",
    "@typescript-eslint/no-array-delete": "error",
    "@typescript-eslint/no-base-to-string": "error",
    "@typescript-eslint/no-duplicate-type-constituents": "error",
    "@typescript-eslint/no-floating-promises": "error",
    "@typescript-eslint/no-for-in-array": "error",
    "@typescript-eslint/no-implied-eval": "error",
    "@typescript-eslint/no-misused-promises": "error",
    "@typescript-eslint/no-redundant-type-constituents": "error",
    "@typescript-eslint/no-unnecessary-type-assertion": "error",
    "@typescript-eslint/no-unsafe-argument": "error",
    "@typescript-eslint/no-unsafe-assignment": "error",
    "@typescript-eslint/no-unsafe-call": "error",
    "@typescript-eslint/no-unsafe-enum-comparison": "error",
    "@typescript-eslint/no-unsafe-member-access": "error",
    "@typescript-eslint/no-unsafe-return": "error",
    "@typescript-eslint/no-unsafe-unary-minus": "error",
    "no-throw-literal": "off",
    "@typescript-eslint/only-throw-error": "error",
    "prefer-promise-reject-errors": "off",
    "@typescript-eslint/prefer-promise-reject-errors": "error",
    "require-await": "off",
    "@typescript-eslint/require-await": "error",
    "@typescript-eslint/restrict-plus-operands": "error",
    "@typescript-eslint/restrict-template-expressions": [
      "warn",
      {
        "allowNumber": true
      }
    ],
    "@typescript-eslint/unbound-method": "error",
    "@typescript-eslint/adjacent-overload-signatures": "error",
    "@typescript-eslint/array-type": [
      "warn",
      {
        "default": "array",
        "readonly": "array"
      }
    ],
    "@typescript-eslint/ban-tslint-comment": "error",
    "@typescript-eslint/consistent-generic-constructors": "error",
    "@typescript-eslint/consistent-indexed-object-style": "error",
    "@typescript-eslint/consistent-type-definitions": "error",
    "@typescript-eslint/no-confusing-non-null-assertion": "error",
    "no-empty-function": "error",
    "@typescript-eslint/no-inferrable-types": "error",
    "@typescript-eslint/non-nullable-type-assertion-style": "error",
    "@typescript-eslint/prefer-for-of": "error",
    "@typescript-eslint/prefer-function-type": "error",
    "@typescript-eslint/prefer-includes": "warn",
    "@typescript-eslint/no-confusing-void-expression": [
      "error",
      {
        "ignoreArrowShorthand": true
      }
    ],
    "@typescript-eslint/no-deprecated": "error",
    "@typescript-eslint/no-dynamic-delete": "error",
    "@typescript-eslint/no-extraneous-class": "error",
    "@typescript-eslint/no-meaningless-void-operator": "error",
    "@typescript-eslint/no-misused-spread": "error",
    "@typescript-eslint/no-mixed-enums": "error",
    "@typescript-eslint/no-non-null-asserted-nullish-coalescing": "error",
    "@typescript-eslint/no-non-null-assertion": "off",
    "@typescript-eslint/no-unnecessary-boolean-literal-compare": "error",
    "@typescript-eslint/no-unnecessary-template-expression": "error",
    "@typescript-eslint/no-unnecessary-type-arguments": "error",
    "no-useless-constructor": "error",
    "@typescript-eslint/prefer-literal-enum-member": "error",
    "@typescript-eslint/prefer-reduce-type-parameter": "error",
    "@typescript-eslint/prefer-return-this-type": "error",
    "@typescript-eslint/related-getter-setter-pairs": "error",
    "@typescript-eslint/return-await": "error",
    "@typescript-eslint/use-unknown-in-catch-callback-variable": "off",
    "no-unassigned-vars": "warn",
    "eqeqeq": [
      "error",
      "always"
    ],
    "no-else-return": "error",
    "@typescript-eslint/explicit-function-return-type": [
      "warn",
      {
        "allowExpressions": true
      }
    ],
    "default-param-last": "warn",
    "no-duplicate-imports": [
      "error",
      {
        "allowSeparateTypeImports": true
      }
    ],
    "@typescript-eslint/no-unsafe-type-assertion": "off",
    "@typescript-eslint/promise-function-async": "error",
    "@typescript-eslint/require-array-sort-compare": "error",
    "@typescript-eslint/switch-exhaustiveness-check": "error",
    "@typescript-eslint/no-unnecessary-condition": "off",
    "@typescript-eslint/prefer-readonly-parameter-types": "off",
    "eslint/max-depth": "off",
    "eslint/no-await-in-loop": "off",
    "import/max-dependencies": "off",
    "import/no-unassigned-import": "off",
    "eslint/max-lines": "off",
    "@typescript-eslint/strict-boolean-expressions": "off",
    "eslint/max-lines-per-function": "off",
    "eslint/sort-vars": "off",
    "eslint/max-classes-per-file": "off",
    "oxc/approx-constant": "off",
    "@stylistic/lines-between-class-members": ["warn", "always", { "exceptAfterSingleLine": true }],
    "@stylistic/quotes": ["warn", "double", {"avoidEscape": true}],
    "@stylistic/space-before-function-paren": [
        "error", {"anonymous": "never", "named": "never", "asyncArrow": "always", "catch": "never"}
    ],
    "@stylistic/comma-spacing": ["error", {"before": false, "after": true}],
    "@stylistic/keyword-spacing": ["warn", {"before": true, "after": true, "overrides": {
        "if": {"after": false},
        "switch": {"after": false},
        "catch": {"after": false},
        "while": {"after": false},
        "for": {"after": false},
    }}],

    "@stylistic/spaced-comment": ["warn", "always", {"markers": [":", "-", "+", "::", "/"]}],
    /* "@stylistic/semi-spacing": "error",
    "@stylistic/max-statements-per-line": ["warn", {"max": 2}],
    "@stylistic/brace-style": ["warn", "stroustrup", {"allowSingleLine": true}],
    "@stylistic/no-extra-semi": "error",
    "@stylistic/no-extra-parens": ["warn", "functions"],
    "@stylistic/semi": ["error", "always"],
    "@stylistic/member-delimiter-style": "warn",
    "@stylistic/no-multiple-empty-lines": ["warn", {"max": 2, "maxEOF": 1, "maxBOF": 0}],
    "@stylistic/no-mixed-spaces-and-tabs": "off",
    "@stylistic/array-bracket-newline": ["warn", "consistent"],
    "@stylistic/array-bracket-spacing": ["warn", "never"],
    "@stylistic/arrow-parens": ["warn", "always"], */
    "@stylistic/yield-star-spacing": ["error", {"after": true, "before": false}],
    "@stylistic/generator-star-spacing": ["error", {"after": true, "before": false}],
    "@stylistic/eol-last": ["error", "always"],
    "@stylistic/new-parens": "error",
    "@stylistic/quote-props": ["warn", "as-needed"],
    "@stylistic/padded-blocks": "off",
    "@stylistic/indent": "off",
    "@stylistic/no-tabs": "off",
    "@stylistic/no-mixed-spaces-and-tabs": "off",

    "sonarjs/no-commented-code": "warn",
    "eslint/no-inline-comments": "off",
    "sonarjs/cognitive-complexity": ["off", 40],
    "sonarjs/void-use": "off",
    "sonarjs/different-types-comparison": "off",
    "sonarjs/no-unenclosed-multiline-block": "off",
    "sonarjs/fixme-tag": "off",
    "sonarjs/no-labels": "off",
    "sonarjs/todo-tag": "off",
    "sonarjs/pseudo-random": "off",
    "sonarjs/no-selector-parameter": "off",
    "sonarjs/no-gratuitous-expressions": "warn",
    "sonarjs/no-empty-collection": "warn",
    "sonarjs/no-unused-collection": "warn",
    "sonarjs/no-use-of-empty-return-value": "warn",
    "sonarjs/no-extra-arguments": "warn",
    "sonarjs/no-redundant-jump": "warn",
  },
  "globals": {
    "document": "readonly",
    "window": "readonly",
    "NodeJS": "readonly",
    "test": "readonly",
    "afterAll": "readonly",
    "beforeAll": "readonly",
    "describe": "readonly",
    "process": "readonly",
    "setTimeout": "readonly",
    "setInterval": "readonly",
    "clearInterval": "readonly",
    "clearTimeout": "readonly",
    "console": "readonly",
    "Buffer": "readonly",
    "structuredClone": "readonly",
    "addEventListener": "readonly",
    "defineProps": "readonly",
    "defineEmits": "readonly",
    "defineModel": "readonly",
    "withDefaults": "readonly",
    "ResizeObserver": "readonly",
	  "Mesh": "readonly"
  }
}
```

### What happened?

This routine (extracted from a bigger source):
```ts
const rgb2lab = (rgb: RGB): LAB => {

    let [r, g, b] = rgb;

    r = (r > 0.04045) ? ((r + 0.055) / 1.055)**2.4 : r / 12.92;
    g = (g > 0.04045) ? ((g + 0.055) / 1.055)**2.4 : g / 12.92;
    b = (b > 0.04045) ? ((b + 0.055) / 1.055)**2.4 : b / 12.92;

    let x = (r * 0.4124 + g * 0.3576 + b * 0.1805) / 0.95047;
    let y = (r * 0.2126 + g * 0.7152 + b * 0.0722) / 1.00000;
    let z = (r * 0.0193 + g * 0.1192 + b * 0.9505) / 1.08883;

    x = (x > 0.008856) ? Math.cbrt(x) : (7.787 * x) + 16/116;
    y = (y > 0.008856) ? Math.cbrt(y) : (7.787 * y) + 16/116;
    z = (z > 0.008856) ? Math.cbrt(z) : (7.787 * z) + 16/116;

    return [(116 * y) - 16, 500 * (x - y), 200 * (y - z)];
};
```
Gives these false positives:
```
  ⚠ eslint(no-useless-assignment): This assigned value is not used in subsequent statements.
    ╭─[src/electron/fingerprint/ContrastingColors.ts:90:9]
 89 │     let y = (r * 0.2126 + g * 0.7152 + b * 0.0722) / 1.00000;
 90 │     let z = (r * 0.0193 + g * 0.1192 + b * 0.9505) / 1.08883;
    ·         ────────────────────────────────────────────────────
 91 │
    ╰────
  help: Consider removing or reusing the assigned value.

  ⚠ eslint(no-useless-assignment): This assigned value is not used in subsequent statements.
    ╭─[src/electron/fingerprint/ContrastingColors.ts:89:9]
 88 │     let x = (r * 0.4124 + g * 0.3576 + b * 0.1805) / 0.95047;
 89 │     let y = (r * 0.2126 + g * 0.7152 + b * 0.0722) / 1.00000;
    ·         ────────────────────────────────────────────────────
 90 │     let z = (r * 0.0193 + g * 0.1192 + b * 0.9505) / 1.08883;
    ╰────
  help: Consider removing or reusing the assigned value.

  ⚠ eslint(no-useless-assignment): This assigned value is not used in subsequent statements.
    ╭─[src/electron/fingerprint/ContrastingColors.ts:88:9]
 87 │
 88 │     let x = (r * 0.4124 + g * 0.3576 + b * 0.1805) / 0.95047;
    ·         ────────────────────────────────────────────────────
 89 │     let y = (r * 0.2126 + g * 0.7152 + b * 0.0722) / 1.00000;
    ╰────
  help: Consider removing or reusing the assigned value.

  ⚠ eslint(no-useless-assignment): This assigned value is not used in subsequent statements.
    ╭─[src/electron/fingerprint/ContrastingColors.ts:82:9]
 81 │
 82 │     let [r, g, b] = rgb;
    ·         ───────────────
 83 │
    ╰────
  help: Consider removing or reusing the assigned value.

  ⚠ eslint(no-useless-assignment): This assigned value is not used in subsequent statements.
    ╭─[src/electron/fingerprint/ContrastingColors.ts:82:9]
 81 │
 82 │     let [r, g, b] = rgb;
    ·         ───────────────
 83 │
    ╰────
  help: Consider removing or reusing the assigned value.

  ⚠ eslint(no-useless-assignment): This assigned value is not used in subsequent statements.
    ╭─[src/electron/fingerprint/ContrastingColors.ts:82:9]
 81 │
 82 │     let [r, g, b] = rgb;
    ·         ───────────────
 83 │
    ╰────
  help: Consider removing or reusing the assigned value.
```
